src/main/java/org/takes/misc/InputStreamsEqual.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.misc;
+
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import org.cactoos.Scalar;
+import org.cactoos.scalar.Not;
+
+/**
+ * Scalar to compare equality of InputStream objects.
+ *
+ * <p>The class is immutable and thread-safe.
+ * @since 2.0
+ */
+public final class InputStreamsEqual implements Scalar<Boolean> {
```

Comment:
@fanifieiev I'm sorry, but I didn't like this approach/classes here. I'm thinks in something like:
```
boolean result = Arrays.equals(
  new InputAsBytes(
    new InputOf(req1.body())
  ).asBytes(),
  new InputAsBytes(
    new InputOf(req2.body())
  ).asBytes(),
)
```
and replace the `InputStreamEqual` with the code above. But if you allow a suggestion, you can get this code (`InputStreamEqual` and  `BytesMatch`) and submit an issue to Cactoos project.

Reply:
@fanifieiev I've been edited my code to a better version. Please, take a look.

Reply:
@fabriciofx `InputAsBytes` is not a public class.
So, that works:
```
new Equality<>(
      new BytesOf(this.body()),
      new BytesOf(other.body())
).value() == 0
```